canvas-peer-grade-calculator/README.md

```diff
@@ -0,0 +1,51 @@
+# canvas-peer-grade-calculator - - README
+
+This project is a deployment wrapper for Longhorn Open Ed Tech's [canvas-peer-grade-calculator](https://github.com/longhornopen/canvas-peer-grade-calculator).
+
+Docker uses an image from the project's container package, [canvaspeergrade](https://github.com/longhornopen/canvas-peer-grade-calculator/pkgs/container/canvaspeergrade).
+
+## Run app locally in Docker
+
+1. Copy `.env.sample` to `.env`…
+    ```shell
+    cp .env.sample .env
+    ```
+2. Edit `.env` contents, using the comments therein as a guide.  Additional
+   information can be found in Longhorn's [README.md](https://github.com/longhornopen/canvas-peer-grade-calculator/blob/6a2ece08b61a16a28f57d841a0399ca99081cbf4/README.md)
+   for their project.
+3. Start the app and db containers with Docker `compose`…
+    ```shell
+    docker compose up --build
+    ```
+4. (Temporary) Run DB migrations.  
+    The following DB migrations need to be applied manually until they can be
+    added to the Docker process and be applied automatically.  These migrations
+    should **_NOT_** be run multiple times with the same database.  The
+    step for creating the session table migration will make **_ANOTHER_**
+    migration, which will **_FAIL_** if it is applied again.
+   * Create a migration for the session database table.  
+     A migration for this was not included in the Docker image provided by
+     Longhorn Open.
+     ```shell
+     docker compose exec app php artisan session:table
```

Comment:
I'd just add this to the `Dockerfile.umich` as

`RUN php artisan session:table`

And remove this documentation bullet point. It's fine to pre-create the files in the container just can't run them. 

I agree that Longhornopen *probably* should do this but the default there is just `SESSION_DRIVER=file`.

Reply:
It's also possibly if you create the migration in the container you won't have the duplicate issue with running a migrations. 

It said for me "Nothing to migrate" when I ran it again. 

However I guess it will cause a problem eventually too since the file is named with the date time `2023_03_02_195645_create_sessions_table`.

So I see what you mean. Maybe we should just create the php files for the tables here and copy them in.

Reply:
I think this is the best way to do it, until they add it upstream.
https://github.com/jonespm/docker-base-images/commit/547cf26e0149c1061bf9e57cfa28db7b6c1c5119

I don't see any way to provide a filename to `artisan:table`, which could be nice. Then just remove these steps. 

You could run the migration at startup but that would need a script [similar to the one I have for UDOIT](https://github.com/jonespm/docker-base-images/blob/main/udoit/config/pre-start.sh). I don't think that's super important but would be nice to add eventually.

Reply:
I was thinking along the same lines.  A script like that would be best, but it's not very important right now.  I'll open a new issue to document that we will need one at some point.  Before a production deployment, I think.

I think Django migrations use conditional SQL statements like, `drop table… if exists`, etc.  These Artisan ones do not.  So I had a case where my DB already had the `session` table, but the code was running in a clean volume without the migration for that table.  So, from the Artisan point of view, it needed to be created.  But then, if I tried to apply the migration, it would fail because the table already existed.  (It's a moot point, I guess, because Artisan will allow us to create as many session table migrations as we like, which also causes errors when applying them.)

Reply:
Okay, I see you created #12, looks good.